### PR TITLE
Add support for sending client certificates

### PR DESCRIPTION
**Pull request by brandur**
Adds support for sending a client certificate to a GitHub Enterprise server.

The client certificate should be encoded with PKCS12 and its path should be set with the configuration option `client_certificate`.

@mislav Can I possibly get a sanity check on this one? I ended up pulling in the PKCS12 package simply because it seems like the most convenient way of encoding a certificate and key. We'd otherwise need to specify the paths to two different files, or stack PEMs into a single file, which I don't think is as widely supported. The nice part about this approach is that something like Mac OS's Keychain makes it quite easy to export a cert/key pair as a `*.p12`.

I'll warn you up front that there is a small possibility of having to add new miscellaneous improvements in the future to help support a more flexible version of this feature. For example, I've required that the PKCS12 file not be passphrase-protected for simplicity, but allowing a passphrase through hub might be desirable for someone else. Others might like to see support for extracting a certificate directly from the Mac OS Keychain (cURL built on Mac supports this) or support for other encodings. It's also possible that no one besides me will ever want or need this option.

Any thoughts? Thanks in advance!

Fixes https://github.com/github/hub/issues/1494.

### CODE REVIEW

File: github/client.go [hunk 1]
```diff
@@ -685,12 +694,53 @@ func (client *Client) newOctokitClient(auth octokit.AuthMethod) *octokit.Client
 	host = normalizeHost(host)
 	apiHostURL := client.absolute(host)
 
-	httpClient := newHttpClient(os.Getenv("HUB_TEST_HOST"), os.Getenv("HUB_VERBOSE") != "")
+	tlsConfig, err := client.newTLSConfig()
+	if err != nil {
+		panic(err)
```
Comment (brandur) [inline github/client.go]: I put this `panic` here as a placeholder, but it may be desirable to refactor this method to return an error.

Comment (e-beach) [reply github/client.go]: There's also `utils.check(err)` which will quit hub if `err` is not nil.

Comment (brandur) [reply github/client.go]: Ah yes, great point! I changed over to use that in f8cb480. Thanks.

**Comment by kandice1978** [comment]
I'm ap confused

**Comment by brandur** [comment]
We ended up designing a different way to work around this problem, so closing out.

**Pull request closed without merging**